apps/prairielearn/src/sync/fromDisk/tags.ts

```diff
@@ -49,19 +46,16 @@ export async function sync(courseId, courseData, questionIds) {
   const res = await sqldb.callOneRowAsync('sync_course_tags', params);
   perf.end('sproc:sync_course_tags');
 
-  /** @type {[string, any][]} */
-  const newTags = res.rows[0].new_tags_json;
+  const newTags = res.rows[0].new_tags_json as [string, any][];
   const tagIdsByName = newTags.reduce((acc, [name, id]) => {
     acc.set(name, id);
     return acc;
-  }, /** @type {Map<String, any>} */ (new Map()));
+  }, new Map<string, any>());
```

Comment:
I know you're just changing types, but:

```suggestion
  const tagIdsByName = Object.fromEntries(newTags);
```

Reply:
This of course changes the type from Map to Object (Record), but this is only used in line 60 in a way where Map does not produce a significant advantage. That said, if you want to keep a Map:

```suggestion
  const tagIdsByName = new Map(newTags);
```

Reply:
Changed in 6934d26.